pkg/identity/ciprovider/principal.go

```diff
@@ -135,10 +143,10 @@ func (principal ciPrincipal) Embed(_ context.Context, cert *x509.Certificate) er
 		s := v.Field(i).String() // value of each field, e.g the template string
 		// We check the field name to avoid to apply the template for the Issuer
 		// Issuer field should always come from the token issuer
-		if s == "" || vType.Field(i).Name == "Issuer" {
+		if strings.Trim(s, " ") == "" || vType.Field(i).Name == "Issuer" {
```

Comment:
Can you use `TrimSpace` instead, which will trim any whitespace characters as well?